cmd/meroxa/root/apps/deploy.go

```diff
@@ -201,10 +202,20 @@ func (d *Deploy) Logger(logger log.Logger) {
 	d.logger = logger
 }
 
+// getAppSource will return the proper destination where the application source will be uploaded and fetched.
+func (d *Deploy) getAppSource(ctx context.Context) (*meroxa.Source, error) {
+	if env := d.flags.Environment; env != "" {
```

Comment:
I'm not sure that I agree about us using v2 only if customers have the feature flag enabled. I think the env info should be entirely handled by the API, whether customers use envs or common

Reply:
@dianadoherty this is referring to the flag provided in the CLI (i.e.: `--env`), not the feature flag.